### PR TITLE
[PM-14] Add workout session deletion

### PR DESCRIPTION
**Pull request by atulb98**
Implement swipe to delete (with undo) for workout sessions

### CODE REVIEW

File: app/src/main/java/io/mochahub/powermeter/workouts/WorkoutFragment.kt [hunk 1]
```diff
@@ -41,12 +47,116 @@ class WorkoutFragment : Fragment() {
...
+        enableSwipeToDelete()
     }
 
     // TODO: Show a dialog fragment with editor where we can CRUD this session
     private fun onWorkoutSessionClicked(workoutSession: WorkoutSession) {
         Toast.makeText(requireContext(), "Clicked: ${workoutSession.date}", Toast.LENGTH_LONG).show()
     }
+
+    private fun enableSwipeToDelete() {
+        val simpleItemTouchCallback = object : ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT) {
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/workouts/WorkoutFragment.kt]: Can you move this into a reusable abstract class so we can use it in the other screens, and override `onSwiped` in each screen :)

https://github.com/power-meter/android-client/pull/12/files#diff-e03ece5c80f80fe203c94520bdea5347

Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/workouts/WorkoutFragment.kt]:
```suggestion
        val simpleItemTouchCallback = object : ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT) {
```

File: app/src/main/res/layout/row_workout.xml [hunk 2]
```diff
@@ -5,9 +5,11 @@
...
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:layout_marginLeft="@dimen/mtrl_card_spacing"
-        android:layout_marginTop="@dimen/mtrl_card_spacing"
-        android:layout_marginRight="@dimen/mtrl_card_spacing"
+        android:layout_marginLeft="0dp"
+        android:layout_marginTop="0dp"
+        android:layout_marginRight="0dp"
+        app:cardCornerRadius="0dp"
+        app:cardElevation="0dp"
```
Comment (brian-norman) [inline app/src/main/res/layout/row_workout.xml]: How come?

**Review by brian-norman — CHANGES_REQUESTED** [review]
Sick delete animation, would like to use it in the rest of the app 😺

File: app/src/main/res/layout/row_workout.xml (hunk 2 above)
Comment (atulb98) [reply app/src/main/res/layout/row_workout.xml]: imo, it looks better flat, but we can change that once the design is finalized for all screens. More of just implementing functionality now anyways

**Review by brian-norman — APPROVED** [review]
🚀

**Pull request merged**